src/client/java/teammates/client/scripts/scalabilitytests/InstructorFeedbackResultsPageScalabilityTest.java

```diff
@@ -18,20 +33,196 @@
+            addQuestions(numQuestions);
+        } else if (testData.feedbackQuestions.size() < numQuestions) { // current number of questions is below required
+            addQuestions(numQuestions - testData.feedbackQuestions.size());
+        } else if (testData.feedbackQuestions.size() > numQuestions) { // current number of questions is above required
+            decreaseNumOfQuestions(numQuestions);
+        }
+
+        // modify set of responses for each test according to its sets of students and questions
+        updateFeedbackResponses();
     }
 
-    private void refreshTestData(String filename) {
-        testData = loadDataBundle(DATA_FOLDER_PATH + filename);
-        removeAndRestoreDataBundle(testData);
+    // verify if entities for testing already exist in datastore
+    private void verifyOrPersistTestDataToDatastore(DataBundle dataToVerify) {
+        log.info("Verifying test data existance in datastore...");
+        Stopwatch stopwatch = new Stopwatch();
+        stopwatch.start();
+
+        for (CourseAttributes course : dataToVerify.courses.values()) {
+            if (getCourse(course) == null) {
+                doPutCourse(course);
+                log.info("Course was added to datastore: " + course);
+            }
+        }
+
+        for (AccountAttributes account : dataToVerify.accounts.values()) {
+            if (getAccount(account) == null) {
+                doPutAccount(account);
+                log.info("Account was added to datastore: " + account);
+            }
+        }
+
+        for (FeedbackSessionAttributes feedbackSession : dataToVerify.feedbackSessions.values()) {
+            if (getFeedbackSession(feedbackSession) == null) {
+                doPutFeedbackSession(feedbackSession);
+                log.info("Feedback Session was added to datastore: " + feedbackSession);
+            }
+        }
+
+        for (InstructorAttributes instructor : dataToVerify.instructors.values()) {
+            if (getInstructor(instructor) == null) {
+                doPutInstructor(instructor);
+                log.info("Instructor was added to datastore: " + instructor);
+            }
+        }
+
+        for (StudentAttributes student : dataToVerify.students.values()) {
+            if (getStudent(student) == null) {
+                doPutStudent(student);
+                log.info("Student was added to datastore: " + student);
+            }
+        }
+
+        for (FeedbackQuestionAttributes question : dataToVerify.feedbackQuestions.values()) {
+            if (getFeedbackQuestion(question) == null) {
+                doPutFeedbackQuestion(question);
+                log.info("Feedback Question was added to datastore: " + question);
+            }
+        }
+
+        int count = 1;
+        for (FeedbackResponseAttributes feedbackResponse : dataToVerify.feedbackResponses.values()) {
+
+            if (getFeedbackResponse(feedbackResponse) == null) {
```

Comment:
Oh I got the answer when working on #2258. The feedback question ID isn't actually 0; it's populated only when it's persisted to the datastore. The `feedbackQuestionId` field in test JSON corresponds to the question number instead (the actual ID is then injected into it before persisting). You'll need to first get the feedback question via session name, course ID and question number (these fields can all be found in the JSON) using the `getFeedbackQuestion` method, read the actual ID from it, then replace the value of the `feedbackQuestionId` field of the 'FeedbackResponseAttributes` object with the actual ID before passing it to `getFeedbackResponse`.